pkg/pvc/pvc_protected_entity.go

```diff
@@ -283,6 +286,67 @@ func (this PVCProtectedEntity) getProtectedEntityForPV(ctx context.Context, pv *
 				return nil, errors.Errorf("Unexpected access mode, %v, for Persistent Volume %s", pv.Spec.AccessModes[0], pv.Name)
 			}
 		}
+	} else if pv.Spec.VsphereVolume != nil {
+		if pv.Spec.AccessModes[0] != core_v1.ReadWriteOnce {
+			return nil, errors.Errorf("Unexpected access mode, %v, for Persistent Volume %s", pv.Spec.AccessModes[0], pv.Name)
+		}
+		if pv.Annotations["pv.kubernetes.io/migrated-to"] == VSphereCSIProvisioner {
+			volumePath := pv.Spec.VsphereVolume.VolumePath
+
+			// Migrated volume must be of type ivd
+			pvPEType := this.getComponentPEType()
+
+			migrationGVR := schema.GroupVersionResource{Group: "cns.vmware.com", Version: "v1alpha1", Resource: "cnsvspherevolumemigrations"}
+			migrationResource := this.ppetm.dynamicClient.Resource(migrationGVR)
+			migrationResourcelist, err := migrationResource.List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return nil, errors.Wrapf(err, "Could not list resource for cnsvspherevolumemigrations")
+			}
+			var pvIDstr string
+			for _, item := range migrationResourcelist.Items {
+				name := item.GetName()
+				// Parse volumeid and volumepath field from cnsvspherevolumemigrations
+				migrationVolumeId, exists, err := unstructured.NestedString(item.UnstructuredContent(), "spec", "volumeid")
+				if err != nil {
+					this.logger.Errorf("Error reading cnsvspherevolumemigrations %s: %v", name, err)
+					continue
+				}
+				if !exists {
+					this.logger.Errorf("Error reading field spec.volumeid cnsvspherevolumemigrations %s: %v", name, err)
+					continue
+				}
+				migrationVolumePath, exists, err := unstructured.NestedString(item.UnstructuredContent(), "spec", "volumepath")
+				if err != nil {
+					this.logger.Errorf("Error reading cnsvspherevolumemigrations %s: %v", name, err)
+					continue
+				}
+				if !exists {
+					this.logger.Errorf("Error reading field spec.volumepath cnsvspherevolumemigrations %s: %v", name, err)
+					continue
+				}
+
+				// If vpath field matches volumePath from migrated volume
+				if migrationVolumePath == volumePath {
+					pvIDstr = migrationVolumeId
+				}
+			}
```

Comment:
Since you are continuing if there are errors, I think you should verify that "pvIDstr" is not an empty string once you finish looping.

Reply:
Added check, 
At the exit of loop, if pvIDstr is empty throw error.